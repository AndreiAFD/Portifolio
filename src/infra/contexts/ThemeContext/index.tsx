import React, { createContext, useState } from "react";
import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../core/theme/theme";

enum ThemeEnum {
  light = "light",
  dark = "dark",
}

interface AppThemeContextProps {
  theme?: string;
  handleChangeTheme?: (theme: ThemeEnum) => void;
}

export const AppThemeContext = createContext<AppThemeContextProps>({});

export const AppThemeContextProvider = ({ children }: { children: any }) => {
  const [actualTheme, setActualTheme] = useState<ThemeEnum>(ThemeEnum.light);

  const chakraTheme = extendTheme({
    colors: {
      ...theme[actualTheme],
    },
  });

  const handleChangeTheme = (nextTheme: ThemeEnum) => {
    setActualTheme(nextTheme);
  };

  return (
    <AppThemeContext.Provider
      value={{
        theme: actualTheme,
        handleChangeTheme,
      }}
    >
      <ThemeProvider theme={theme[actualTheme] as typeof theme["dark"]}>
        <ChakraProvider theme={chakraTheme}>{children}</ChakraProvider>
      </ThemeProvider>
    </AppThemeContext.Provider>
  );
};
