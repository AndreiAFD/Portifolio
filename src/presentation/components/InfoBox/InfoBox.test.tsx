import React from "react";
import { render, screen } from "@testing-library/react";
import { InfoBox } from ".";
import { AppThemeContextProvider } from "../../../infra/contexts/ThemeContext";

const ComponentToTest = () => {
  return (
    <AppThemeContextProvider>
      <InfoBox />
    </AppThemeContextProvider>
  );
};

describe("InfoBox component", () => {
  it("should render correctly", () => {
    render(<ComponentToTest />);
    const container = screen.getByTestId("infobox-container-testid");
    expect(container).toMatchSnapshot();
  });
});
