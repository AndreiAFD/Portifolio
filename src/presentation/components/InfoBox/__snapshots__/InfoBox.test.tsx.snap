// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox component should render correctly 1`] = `
<div
  class="css-8cgegm"
  data-testid="infobox-container-testid"
>
  <div
    class="css-1lads1q"
  >
    <span
      class="chakra-avatar css-1ni48nu"
    >
      <svg
        aria-label=" avatar"
        class="chakra-avatar__svg css-16ite8i"
        role="img"
        viewBox="0 0 128 128"
      >
        <path
          d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
          fill="currentColor"
        />
        <path
          d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
          fill="currentColor"
        />
      </svg>
    </span>
  </div>
  <div
    class="css-iuw0g7"
  >
    <p
      class="chakra-text css-1xqtn9i"
    >
      Andrei Amaral
    </p>
    <p
      class="chakra-text css-63qbpm"
      data-testid="badge-text-testid"
    >
      Disponível
    </p>
  </div>
  <p
    class="chakra-text css-1hen3ta"
  >
    Desenvolvedor frontend autônomo apaixonado por tecnologia.
  </p>
  <div
    class="css-17xy7xo"
  >
    <p
      class="chakra-text css-1hc4vyr"
    >
      Informações de contato
    </p>
  </div>
  <div
    class="css-13v6rjs"
  >
    <p
      class="chakra-text css-1drml6o"
    >
      Email:
    </p>
    <p
      class="chakra-text css-1u1uxt6"
    >
      andrei.frontdev@gmail.com
    </p>
  </div>
  <div
    class="css-1sbrxma"
  >
    <p
      class="chakra-text css-1drml6o"
    >
      Celular e WhatsApp:
       
    </p>
    <p
      class="chakra-text css-6xix1i"
    >
      +55 11 99240 5312 
    </p>
  </div>
  <div
    class="css-1d5veyz"
  >
    <button
      class="chakra-button css-1y6ftdq"
      data-testid="badgelink-container-testid"
      type="button"
    >
      <a
        class="chakra-link css-3zcpgt"
        data-testid="badgelink-testid"
        href="https://www.linkedin.com/in/andrei-amaral-327493191/"
        target="_blank"
      >
        <img
          class="chakra-image css-19xcex3"
          src="linkedin.png"
        />
        Linkedin
      </a>
    </button>
    <button
      class="chakra-button css-1uv1qke"
      data-testid="badgelink-container-testid"
      type="button"
    >
      <a
        class="chakra-link css-3zcpgt"
        data-testid="badgelink-testid"
        target="_blank"
      >
        <img
          class="chakra-image css-19xcex3"
          src="whatsapp.png"
        />
        WhatsApp
      </a>
    </button>
  </div>
</div>
`;
